### PR TITLE
[12.0][FIX] github_connector: Add credentials to clone a private repository.

### PR DESCRIPTION
**Pull request by imanie383**
When you try to clone a private repository, the process is interrupted by the request for credentials like in the following video : https://youtu.be/7tDVTRsqWas

### CODE REVIEW

**Comment by pedrobaeza** [comment]
This should be done through a token, not with credentials. Why do you need to add this?

**Comment by pedrobaeza** [comment]
@CarlosRoca13 can you check this? Specifically, if it's flawing in the same vulnerability than previously.

File: github_connector/models/github_repository_branch.py
Comment (yajo) [inline github_connector/models/github_repository_branch.py]: According to [docs](https://docs.github.com/en/developers/apps/authenticating-with-github-apps#http-based-git-access-by-an-installation), this should be:

```suggestion
            login = "x-access-token:%s@" % (tools.config.get('github_token'))
```

**Review by yajo — CHANGES_REQUESTED** [review]

**Comment by yajo** [comment]
The need is obvious by the video: to clone private github repos.
It doesn't make much sense that the token is global, as different repos might need different tokens. But well... that's how it is designed, and we don't need to change that until it's really needed.
You should explain in the README which scopes does the token need to work fine. Also you should explain that the git code should be found in a secure storage where only the Odoo process has read and write capabilities, or secrets might get exposed.
Since the login/password is stored in the config file, and that file needs more or less the same access level, this is not exposing a new vulnerability if it is configured properly, but that's why you should add those warnings to the README.
Also you should recommend using tokens instead of login+password, although I think that's already done elsewhere.

**Comment by imanie383** [comment]
> The need is obvious by the video: to clone private github repos.
> It doesn't make much sense that the token is global, as different repos might need different tokens. But well... that's how it is designed, and we don't need to change that until it's really needed.
> You should explain in the README which scopes does the token need to work fine. Also you should explain that the git code should be found in a secure storage where only the Odoo process has read and write capabilities, or secrets might get exposed.
> Since the login/password is stored in the config file, and that file needs more or less the same access level, this is not exposing a new vulnerability if it is configured properly, but that's why you should add those warnings to the README.
> Also you should recommend using tokens instead of login+password, although I think that's already done elsewhere.

Done @Yajo

**Comment by hugho-ad** [comment]
@Yajo 

could you review please?

File: github_connector/README.rst
Comment (yajo) [inline github_connector/README.rst]: Do you really need full write access here? And access to accept or decline invitations? Isn't that too much?

File: github_connector/README.rst
Comment (yajo) [inline github_connector/README.rst]:
```suggestion
* For security, the git code (source_code_local_path) and the config file (odoo.conf) should be found in a secure storage where only the Odoo process has read and write capabilities.
Otherwise, the code and token (or login/password if you use those -- don't do it!) may be exposed.
```

File: github_connector/README.rst
Comment (imanie383) [reply github_connector/README.rst]: I tried everyone else but, it does not work

https://youtu.be/07F2pHfvHvA

Also, although the token has full access, these depend on  the user's permissions has in each repo

**Review by hugho-ad — APPROVED** [review]

File: github_connector/README.rst
Comment (yajo) [reply github_connector/README.rst]: That's a clear answer, thanks! :wink:

File: github_connector/README.rst
Comment (yajo) [inline github_connector/README.rst]: You need to add these docs in the corresponding section inside the `readme` directory. Otherwise, the bot will rebuild the README and destroy this change after merge.

**Review by yajo — CHANGES_REQUESTED** [review]

File: github_connector/README.rst
Comment (pedrobaeza) [reply github_connector/README.rst]: Isn't interesting to put the video link or this explanation for avoiding further questions in the future?

**Comment by yajo** [comment]
Yes, good idea.

File: github_connector/README.rst
Comment (imanie383) [reply github_connector/README.rst]: Done @Yajo

File: github_connector/README.rst
Comment (imanie383) [reply github_connector/README.rst]: Where?

File: github_connector/readme/USAGE.rst
Comment (yajo) [inline github_connector/readme/USAGE.rst]: Use this to avoid having to update the link manually on each version bump:

```suggestion
   .. image:: ../static/description/generate_new_token.png
```

The bot will convert that to an absolute path when building the README.

File: github_connector/readme/USAGE.rst
Comment (yajo) [inline github_connector/readme/USAGE.rst]:
```suggestion
   .. image:: ../static/description/personal_access_token.png
```

File: github_connector/readme/USAGE.rst
Comment (yajo) [inline github_connector/readme/USAGE.rst]:
```suggestion
Add one of these scopes to your token:

* **(no scope)** : Grants read-only access to public information (includes public user profile info, public repository info, and gists).
* **repo** : Grants full access to private and public repositories. `See here why is this needed <https://youtu.be/07F2pHfvHvA>`__.
```

**Review by yajo — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Please forward-port it to v13

**Pull request merged**
